File: src/test/java/application/browser/GraveyardHelper.java
package application.browser;

import org.openqa.selenium.By;
import org.openqa.selenium.WebElement;

import com.matag.player.PlayerType;

public class GraveyardHelper extends AbstractCardContainerHelper {

  private final PlayerType playerType;

  GraveyardHelper(MatagBrowser matagBrowser, PlayerType playerType) {
    super(matagBrowser);
    this.playerType = playerType;
  }

  @Override
  protected WebElement containerElement() {
    if (playerType == PlayerType.PLAYER) {
      return matagBrowser.findElement(By.id("player-graveyard"));
    } else {
      return matagBrowser.findElement(By.id("opponent-graveyard"));
    }
  }
}

File: src/test/kotlin/application/browser/GraveyardHelper.kt
package application.browser

import com.matag.player.PlayerType
import org.openqa.selenium.By
import org.openqa.selenium.WebElement

class GraveyardHelper internal constructor(matagBrowser: MatagBrowser?, private val playerType: PlayerType?) :
    AbstractCardContainerHelper(matagBrowser) {
    override fun containerElement(): WebElement? {
        if (playerType == PlayerType.PLAYER) {
            return matagBrowser.findElement(By.id("player-graveyard"))
        } else {
            return matagBrowser.findElement(By.id("opponent-graveyard"))
        }
    }
}
